app/webutil.js

```diff
@@ -211,18 +221,16 @@ export function injectParamIfMissing (path, param, value) {
 // will receive either an event or an error on failure.
 export function fetchJSON(path, resolve, reject) {
     // NB: IE11 doesn't support JSON as a responseType
-    var req = new XMLHttpRequest();
+    const req = new XMLHttpRequest();
     req.open('GET', path);
 
     req.onload = function () {
         if (req.status === 200) {
             try {
-                var resObj = JSON.parse(req.responseText);
+                resolve(JSON.parse(req.responseText));
```

Comment:
This is a change of behaviour as it will grab any exceptions from the resolve handler as well. Please keep it the previous way.

Reply:
`resolve(JSON.parse(req.responseText));` is functionally equivalent to:
```
const json = JSON.parse(req.responseText);
resolve(json);
```
since the parse method is executed first and it's result passed to the resolve method call.
If `JSON.parse` throws an exception, the resolve method is never called and the try-catch with capture the exception.

Do you still prefer the second version with the extra variable?

Reply:
As @CendioOssman is hinting, we should try avoid any changes in behavior. Imagine troubleshooting or changing something related to this code in the future, and jumping between versions and checking blame, this commit could easily become confusing.